docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -106,9 +105,9 @@ final outcome depends on how index records are stored in existing index files.
 Two or more index files can be merged with each other, if and only if they all have index records which belong to the
 same bucket (according to index configuration).
 When running `optimizeIndex` command, Hyperspace tries to find such groups of index files and merge them together.
-However, if most of the index files have records from different buckets, then final physical layout of index files
-after running optimize would be similar to the original layout. An example of such an index is an index right after
-creation or full refresh, or an index which is refreshed incrementally with very few additional small source data files.    
+However, if for each given bucket, all index records are already stored in a single index file (i.e. we dont have several index
+files for a bucket), then final physical layout of index files after running optimize would be similar to the original
```

Comment:
I think it should be the same if all index records are already stored in a single index file for each bucket, we shouldn't change it. (Though current optimizeIndex rewrites the data and change it, but we don't need to describe it here.) and "final physical layout" term could be confusing. 

How about just removing "However ~~~ original layout" lines?

Reply:
I revised that part again. Plz take a look at it.